pkg/webhooks/jobset_webhook.go

```diff
@@ -11,30 +11,38 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha2
+package webhooks
```

Comment:
Could we do this PR in stages?

There is a lot of changes here but its mixed with some cleanup/imports.

Maybe let's move the pkg and change the imports first.

And then we add the validation?

Reply:
Sounds good, what do you think of the concept/approach though? And about adding https://github.com/kubernetes-sigs/kubectl-validate as a dependency?

Reply:
I think its an elegant solution. I think the dependency increase is probably going from including apiserver.

Reply:
Updated this PR to only focus on migrating jobset webhook to webhooks pkg and very minor cleanup